libs/core/src/main/java/org/opensearch/core/index/Index.java

```diff
@@ -134,6 +136,15 @@ public static Index fromXContent(final XContentParser parser) throws IOException
         return INDEX_PARSER.parse(parser, null).build();
     }
 
+    public void writeToIndexOutput(IndexOutput out) throws IOException {
```

Comment:
This component already implements Writeable and can serialize/deserialize with StreamInput/Output.  It looks like we are missing something here in the remote implementation if we have to update these objects.  I think the issue here is VersionedCodecStreamWrapper should be writing/reading using `IndexOutputOutputStream` and `InputStreamIndexInput` and the individual objects also implement Writeable.  Not sure if there is a compelling reason this component isn't using this already.

For the sake of this PR - I think the right thing to do is not use these specific methods and write/construct the entire checkpoint inside of `RemoteSegmentMetadata` until we refactor.

Reply:
Updated.